processor/useragentprocessor/useragentprocessor.go

```diff
@@ -0,0 +1,85 @@
+package useragentprocessor
+
+import (
+	"context"
+
+	"github.com/ua-parser/uap-go/uaparser"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+type userAgentProcessor struct {
+	config       *Config
+	logger       *zap.Logger
+	uaParser     *uaparser.Parser
+	nextConsumer consumer.TracesConsumer
+}
+
+func (u *userAgentProcessor) ConsumeTraces(ctx context.Context, td pdata.Traces) error {
+	rss := td.ResourceSpans()
+	for i := 0; i < rss.Len(); i++ {
+		rs := rss.At(i)
+		if rs.IsNil() {
+			continue
+		}
+		ilss := rss.At(i).InstrumentationLibrarySpans()
+		for j := 0; j < ilss.Len(); j++ {
+			ils := ilss.At(j)
+			if ils.IsNil() {
+				continue
+			}
+			spans := ils.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+				if span.IsNil() {
+					// Do not create empty spans just to add attributes
+					continue
+				}
+				if attr, ok := span.Attributes().Get(u.config.UserAgentTag); ok {
+					u.enrichSpanWithUserAgent(attr.StringVal(), span)
+				}
+			}
+		}
+	}
+	return u.nextConsumer.ConsumeTraces(ctx, td)
+}
+
+func (u *userAgentProcessor) Start(context.Context, component.Host) error {
+	return nil
+}
+
+func (u *userAgentProcessor) Shutdown(context.Context) error {
+	return nil
+}
+
+func (u *userAgentProcessor) GetCapabilities() component.ProcessorCapabilities {
+	return component.ProcessorCapabilities{MutatesConsumedData: true}
+}
+
+func (u *userAgentProcessor) enrichSpanWithUserAgent(userAgent string, span pdata.Span) {
+	c := u.uaParser.Parse(userAgent)
+	span.Attributes().Insert("browserFamily", pdata.NewAttributeValueString(c.UserAgent.Family))
+	span.Attributes().Insert("browserVersion", pdata.NewAttributeValueString(c.UserAgent.ToVersionString()))
+	span.Attributes().Insert("deviceFamily", pdata.NewAttributeValueString(c.Device.Family))
+	span.Attributes().Insert("deviceBrand", pdata.NewAttributeValueString(c.Device.Brand))
+	span.Attributes().Insert("deviceModel", pdata.NewAttributeValueString(c.Device.Model))
+	span.Attributes().Insert("osFamily", pdata.NewAttributeValueString(c.Os.Family))
+	span.Attributes().Insert("osVersion", pdata.NewAttributeValueString(c.Os.ToVersionString()))
+}
```

Comment:
Some of the conventions are defined here:
https://github.com/open-telemetry/opentelemetry-collector/blob/master/translator/conventions/opentelemetry.go#L19

Some are missing in the specification (probably need to add them):
https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#todos

Reply:
Also they should be part of the "Resource" not Span :)

Reply:
@bogdandrutu thank you for the feedback. Moved the logic to look for usrAgent in `Resource` and added the new attributes in that section.
Do you recommend me to propose the new tags under `conventions` first in a separate PR?

Reply:
To make progress, I think we should reduce the scope of this PR to only elements that are defined in the specs, then once we have specs for others we add them. What do you think? This way I think we don't block progress and we allow future improvements

Reply:
Sounds good. Under conventions, i currently see couple of them `os.type` and `os.description`. 
https://github.com/open-telemetry/opentelemetry-collector/blob/master/translator/conventions/opentelemetry.go#L56-L57

Could you please recommend the right forum where we can discuss about adding new fields to the specs?

Reply:
> > @tannaga please sign the CLA and make sure EasyCLA passes.
> 
> Couple of my commits were made using wrong email address. I am trying to use rebase and amend author options, but I am running into all sort of conflicts in `go.sum`. Do you know if there is an easy way to fix the email address for the 2 commits?

I tried few things to fix the author of my earlier conflicts, but it seems a little messy. I will close this PR and submit a new one.
cc @bogdandrutu @tigrannajaryan